arbor/backends/gpu/matrix_state_fine.hpp

```diff
@@ -345,12 +344,15 @@ struct matrix_state_fine {
             }
             auto prev_end = levels_start.back();
             levels_start.push_back(prev_end + branch_map.size());
-            data_size.push_back(pos);
+            data_partition.push_back(pos);
         }
 
         // set matrix state
         matrix_size = p.size();
 
+	// set data partition size
+	data_size = data_partition.back();
```

Comment:
Instead you can set `data_size = pos;` immediately after the loop on line349.

Reply:
Done.